mmengine/hooks/benchmark_hook.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from mmengine.registry import HOOKS
+from .hook import Hook
+
+
+@HOOKS.register_module()
+class BenchmarkHook(Hook):
+    """A hook that logs the training speed of each epch for benchmark."""
+
+    priority = 'NORMAL'
+
+    def after_train_epoch(self, runner) -> None:
+        """We use the average throughput in iterations of each training epcoh
+        and skip the first 50 iterations of each epoch to skip GPU warmup time.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        message_hub = runner.message_hub
+        max_iter_num = len(runner.train_dataloader)
+        speed = message_hub.get_scalar('train/time').mean(max_iter_num - 50)
```

Comment:
The number "50" seems a bit magic, If `max_iter_num` is smaller than 50, it will get an unexpected result. Since we want to statistic the consumed time of an epoch, why do we still need to skip the first 50 iterations?

Reply:
Generally we skip the first 50 iterations of each epoch to skip GPU warmup time. I think if `max_iter_num` is smaller than 50, we could just use the `max_iter_num` for calculation, please @ZwwWayne  have a look at this.